src/Provider/GoogleMaps/GoogleMaps.php

```diff
@@ -264,9 +264,42 @@ private function fetchUrl(string $url, string $locale = null, int $limit, string
             if (isset($result->formatted_address)) {
                 $address = $address->withFormattedAddress($result->formatted_address);
             }
+            if ($builder->hasValue('streetAddress')) {
```

Comment:
Because you cannot do `$address->getPointOfInterest()`;

Also dynamic properties like `$new->$name = $value;` is bad for performance.